### ISSUE TITLE
Permissions for using certain endpoints when creating courses

### ISSUE THREAD

**Issue by jathak** [okpy/ok#1322]
Right now, anyone can create a course with any available endpoint.

This means that a course's standard endpoint format may already have been taken for a particular semester. We should have some system of reserving certain patterns of endpoints for certain users.

**Comment by colinschoen** [comment]
Makes sense. Has this happened for an established course? We can probably manually fix if so.

**Comment by jathak** [comment]
There were empty (aside from the creator) courses for cal/cs61a/sp18, cal/cs61a/su18, and cal/cs61a/fa18 that I used admin permissions to remove the creator and repurpose them for 61A, but if someone ever creates a course that's actually being used with one of our endpoints, that would be an issue (since changing their endpoint in the middle of the course would break links for students)

**Comment by Sumukh** [comment]
👍 this is a good idea.

**Comment by Sumukh** [comment]
we should also change the default value to not suggest it being some variant of cal/cs61a/sp18
